skops/io/_general.py

```diff
@@ -280,7 +318,7 @@ def __init__(self, state, load_context: LoadContext, trusted=False):
             "step": state["content"]["step"],
         }
 
-    def _construct(self):
+    def _construct(self) -> Any:
```

Comment:
This comment applies to basically all of the `_construct` functions:

This always return a `slice`, so shouldn't we change the type to ` -> slice:`?

Reply:
I actually considered that, but decided against it.

The reason is that those types would not help us in our code. That's because everywhere we call `_construct`, we cannot statically know which of the dozens of types it would return. And this is because we work with generic `state`s. If we had type-specific states, like `DictState` or `NdArrayState`, we may be able to add overloads that say what type is returned by `_construct` based on the state, but we don't have that (and it would be difficult to implement, maybe even impossible with the current typing situation).

So if we added return type annotations here, the only thing that mypy could infer is that `_construct` returns a `Union[str, int, float, dict, list, NDArray, ...]`, which is not more useful than `Any` in practice.

Reply:
In that case, why do we then need to have the type annotation in the first place? can't we just remove it?